service/controller/resource/tcnp/create.go

```diff
@@ -716,3 +737,83 @@ func idFromGroups(groups []*ec2.SecurityGroup, name string) string {
 
 	return ""
 }
+
+func isTCCPNUpdated(ctx context.Context, cr infrastructurev1alpha2.AWSMachineDeployment) (bool, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	// check if TCCPN CF stack is updated
+	{
+		i := &cloudformation.DescribeStacksInput{
+			StackName: aws.String(key.StackNameTCCPN(&cr)),
+		}
+
+		o, err := cc.Client.TenantCluster.AWS.CloudFormation.DescribeStacks(i)
```

Comment:
We have an internal package for the cloudformation magic. Can you check of we can reuse that? Then we would not have to write all that code again in here.

Reply:
I tried to use it